src/Simplex_tree/include/gudhi/Simplex_tree/multi_filtrations/Line.h

```diff
@@ -0,0 +1,233 @@
+/*    This file is part of the Gudhi Library - https://gudhi.inria.fr/ - which
+ * is released under MIT. See file LICENSE or go to
+ * https://gudhi.inria.fr/licensing/ for full license details. Author(s): David
+ * Loiseaux
+ *
+ *    tCopyright (C) 2023 Inria
+ *
+ *    Modification(s):
+ *      - YYYY/MM Author: Description of the modification
+ */
+
+#ifndef LINE_FILTRATION_TRANSLATION_H_INCLUDED
+#define LINE_FILTRATION_TRANSLATION_H_INCLUDED
+
+#include "Box.h"
+#include "Finitely_critical_filtrations.h"
+#include <cstddef>
+
+namespace Gudhi::multiparameter::multi_filtrations {
+
+template <typename T> class Line {
+public:
+  using point_type = One_critical_filtration<T>;
+  using kcritical_point_type = Multi_critical_filtration<T>;
+  Line();
+  Line(const point_type &x);
+  Line(point_type &&x);
+  Line(const point_type &x, const point_type &v);
+  inline point_type push_forward(point_type x) const;
+  template <typename U=T>
+  inline U push_forward2(const point_type &x) const;
+  template <typename U=T>
+  inline U push_forward2(const kcritical_point_type &x) const;
+  inline point_type push_back(point_type x) const;
+  template <typename U=T>
+  inline U push_back2(const point_type &x) const;
+  template <typename U=T>
+  inline U push_back2(const kcritical_point_type &x) const;
+  inline int get_dim() const;
+  std::pair<point_type, point_type> get_bounds(const Box<T> &box) const;
+  std::pair<T,T> get_bounds2(const Box<T> &box) const;
+
+  // translation
+  inline friend Line &operator+=(Line &to_translate, const point_type &x) {
+    to_translate.basepoint_ -= x;
+    return to_translate;
+  }
+
+  inline point_type &basepoint() { return basepoint_; }
+  inline point_type &direction() { return direction_; }
+  inline const point_type &basepoint() const { return basepoint_; }
+  inline const point_type &direction() const { return direction_; }
+
+private:
+  point_type basepoint_; // any point on the line
+  point_type direction_; // direction of the line
+};
+template <typename T> Line<T>::Line() {}
+
+template <typename T> Line<T>::Line(const point_type &x) : basepoint_(x) {}
+template <typename T>
+Line<T>::Line(point_type &&x) : basepoint_(std::move(x)) {}
+template <typename T>
+Line<T>::Line(const point_type &x, const point_type &v)
+    : basepoint_(x), direction_(v) {}
+template <typename T>
+inline typename Line<T>::point_type
+Line<T>::push_forward(point_type x) const { // TODO remove copy
```

Comment:
IIUC, x is interpreted as the base point of an upper-right quadrant (a 1-critical thing), the line has a built-in parametrization, and this function returns the parameter of the (smallest) intersection point between the line and the quadrant? I'd like to see comments like that so I don't need to read the code...
(I assume this is called push_forward because we can start from x and increase ("forward") its coordinates as little as possible to bring it to the line ("project" more than "push"?))

Reply:
Yes, sorry, I'll add some comments later; this is exactly it. `project` seems great to me. There is also the `push_back` method that does the same thing but with the opposite quadrant; so `project` won't be enough.